incubator/virtualcluster/pkg/syncer/syncer.go

```diff
@@ -262,12 +262,18 @@ func (s *Syncer) addCluster(key string, vc *v1alpha1.Virtualcluster) error {
 
 	clusterName := conversion.ToClusterKey(vc)
 
-	adminKubeConfigSecret, err := s.secretClient.Secrets(clusterName).Get(KubeconfigAdmin, metav1.GetOptions{})
-	if err != nil {
-		return fmt.Errorf("failed to get secret (%s) for virtual cluster in root namespace %s: %v", KubeconfigAdmin, clusterName, err)
+	var adminKubeConfigBytes []byte
+	if adminKubeConfig, exists := vc.GetAnnotations()[constants.LabelAdminKubeConfig]; exists {
+		adminKubeConfigBytes = []byte(adminKubeConfig)
```

Comment:
contents of kubeconfig is encoded in base64 format. When getting adminKubeConfig from annotation, syncer need to decode the []byte.